crates/stellar-scaffold-cli/src/commands/build/clients.rs

```diff
@@ -440,98 +463,154 @@ export default new Client.Client({{
-                eprintln!("📲 installing {name:?} wasm bytecode on-chain...");
-                let hash = cli::contract::upload::Cmd::parse_arg_vec(&[
-                    "--wasm",
-                    wasm_path
-                        .to_str()
-                        .expect("we do not support non-utf8 paths"),
-                ])?
-                .run_against_rpc_server(None, None)
-                .await?
-                .into_result()
-                .expect("no hash returned by 'contract install'")
-                .to_string();
-                eprintln!("    ↳ hash: {hash}");
-
-                // Check if we have an alias saved for this contract
-                let alias = Self::get_contract_alias(&name, workspace_root)?;
-                if let Some(contract_id) = alias {
-                    match self
-                        .contract_hash_matches(&contract_id, &hash, network, workspace_root)
-                        .await
-                    {
-                        Ok(true) => {
-                            eprintln!("✅ Contract {name:?} is up to date");
-                            continue;
-                        }
-                        Ok(false) => eprintln!("🔄 Updating contract {name:?}"),
-                        Err(e) => return Err(e),
-                    }
-                }
+        Ok(())
+    }
 
-                eprintln!("🪞 instantiating {name:?} smart contract");
-                let new_contract_id = cli::contract::deploy::wasm::Cmd::parse_arg_vec(&[
-                    "--alias",
-                    &name,
-                    "--wasm-hash",
-                    &hash,
-                ])?
-                .run_against_rpc_server(None, None)
-                .await?
-                .into_result()
-                .expect("no contract id returned by 'contract deploy'");
-                eprintln!("    ↳ contract_id: {new_contract_id}");
-
-                // Save the alias for future use
-                Self::save_contract_alias(&name, &new_contract_id, network, workspace_root)?;
-
-                new_contract_id
-            };
-
-            // Run init script if we're in development or test environment
-            if env == "development" || env == "testing" {
-                if let Some(settings) = settings {
-                    if let Some(init_script) = &settings.init {
-                        eprintln!("🚀 Running initialization script for {name:?}");
-                        self.run_init_script(&name, &contract_id, init_script)
-                            .await?;
-                    }
+    async fn process_single_contract(
+        &self,
+        name: &str,
+        workspace_root: &std::path::Path,
+        contracts: Option<&IndexMap<Box<str>, env_toml::Contract>>,
+        network: &Network,
+        env: &str,
+    ) -> Result<(), Error> {
+        let settings = match contracts {
+            Some(contracts) => contracts.get(name),
+            None => None,
+        };
```

Comment:
Better to use a map. However, as I said above it's better to just pass the map type since it has a default and you can use it directly. Furthermore it looks like it in this example the map is passed to this method only to use a parameter to access the value in the map. So IMO it's the caller's responsibility to provide the value. In this case it should be unwrapped or default so that there is a concrete type for settings.
```suggestion
        let settings = contracts.map(|c| c.get(name));
```

Reply:
I moved it to the caller and pass the settings. Although I ended up using this to get the settings since what you showed returned an Option<Option<&Contract>>. 
```
            let settings = contracts
                .and_then(|contracts| contracts.get(name.as_str()))
                .cloned()
                .unwrap_or_default();
```